### PR TITLE
Feature/engine api override

### LINKED ISSUE #4172: Implement override to force engine-api in the absence of TTD
### Description
In order to smooth the preparations for the merge, consensus is that we want execution clients to enable the engine API even in the absence of merge parameters.  This is in essence a re-enabling of the `--engine-rpc-enabled` api flag removed in #3875

### Acceptance Criteria
* make the engine-api available in the absence of merge configs
* ensure rational behavior pre-TTD calls to engine api endpoints

### PR DESCRIPTION
**Pull request by garyschulte**
## PR description
Allow besu to have the engine api enabled, when there are not merge configs.  includes:
* un-hiding/un-deprecating `--engine-rpc-enabled`
* remove old isMergeEnabled checks in a few block header validators
* use merge-specific block header validator, rather than appending to MainnetBlockHeaderValidator
* disable all engine api methods except engine_exchangeTransitionConfigurationV1 in the absence of an engine-api-compatible coordinator
* add "placeholder" TTD in the absence of a TTD configuration

Tested with Hive and Kurtosis since this affects block validation.  Both are 🟢 

## Fixed Issue(s)

fixes #4172 

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by daniellehrner** [comment]
Just today there was a discussion on Discord that revealed that all CL clients already use a placeholder TTD for mainnet. So I guess the easiest would be to just use that one instead:

https://discord.com/channels/595666850260713488/892088344438255616/1002018160268038164

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java [hunk 1]
```diff
@@ -166,10 +166,10 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
     final var block =
         new Block(newBlockHeader, new BlockBody(transactions, Collections.emptyList()));
 
-    if (mergeContext.isSyncing() || parentHeader.isEmpty()) {
+    if (mergeContext.map(c -> c.isSyncing()).orElse(Boolean.TRUE) || parentHeader.isEmpty()) {
```
Comment (jflo) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java:169]: This is kinda hard to read. It looks like it evaluates to a no-op, does anyone else find this harder to read?

**Review by jflo — APPROVED** [review]
One thing that i found hard to parse, but otherwise this cleans up a ton of conditionals and class casting.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java (hunk 1 above)
Comment (garyschulte) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java:169]: I can see how it is harder to read, it is a side effect of making mergeContext an Optional.  looking for an approach that is easier on the 👀

Comment (garyschulte) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java]: Supplier/function reference ftw 👍
